### DISCUSSION TITLE
Code models score low on humaneval dataset

### DISCUSSION THREAD

**Discussion by PJY-coder** [open-compass/opencompass#1067] [Q&A] [answered]
我使用opencompass测试了starcoder2，wizardcoder，codellama等代码大模型，在humaneval上的评分非常低(小于10)，为什么它们的预处理方式都是一样的呢？不同的代码大模型前后处理好像差别还挺大的吧？

请问大家有遇到这个情况吗？

**Comment by jingmingzhuo** [comment]
Could you provide the configs of models and datasets for testing? Thanks!

**Reply by PJY-coder** [reply]
Just use the default config file given in 'opencompass/configs/models/', such as "opencompass/configs/models/wizardcoder/hf_wizardcoder_15b.py" and "opencompass/configs/models/starcoder2/hf_starcoder2_15b.py". Only change the model path to local directory.

The commad I ran:  python3 run.py --models hf_wizardcoder_15b --datasets humaneval_gen

**Reply by jingmingzhuo** [reply]
For starcoder2, you may need to use a base config, such as configs/datasets/humaneval/humaneval_gen_fd5822.py, because its a base model without going through the SFT process,  refer to [this](https://github.com/bigcode-project/starcoder2/issues/9).

**Reply by PJY-coder** [reply]
thx,I'll try starcoder2 again. 
How about wizardcoder-15b?

**Reply by jingmingzhuo** [reply]
Sorry for taking so long to respond.

I checked the config of wizardcoer and and found that it lacks a meta template. Here is a modified version:

```python
from opencompass.models import HuggingFaceCausalLM

_meta_template = dict(
    round=[
        dict(role="HUMAN", begin='### Instruction:\n', end='\n\n'),
        dict(role="BOT", begin="### Response:\n", end='</s>', generate=True),
    ],
)

models = [
    # WizardCoder 15B
    dict(
        type=HuggingFaceCausalLM,
        abbr='WizardCoder-15B-V1.0',
        path="WizardLM/WizardCoder-15B-V1.0",
        tokenizer_path='WizardLM/WizardCoder-15B-V1.0',
        tokenizer_kwargs=dict(
            padding_side='left',
            truncation_side='left',
            trust_remote_code=True,
        ),
        meta_template=_meta_template,
        max_out_len=1024,
        max_seq_len=2048,
        batch_size=8,
        model_kwargs=dict(trust_remote_code=True, device_map='auto'),
        run_cfg=dict(num_gpus=2, num_procs=1),
    ),
]
```

In my testing, its pass@1 score under humaneval_gen is 47.56.

**Reply by PJY-coder** [reply]
Thanks for answering, but I think the template u provided is not correct. 

According to [here](https://github.com/nlpxucan/WizardLM/tree/main/WizardCoder,) the pass@1 score of humaneval should be around 55. And I further discovered that  humaneval_gen_4a6eef works well, which is same with [this](https://github.com/nlpxucan/WizardLM/blob/main/WizardCoder/src/humaneval_gen.py). 

In my testing, its pass@1 score under humaneval_gen is 54.27 with accelerator vllm, 52.43 without accelerator vllm.  Even though the number are acceptable, but there still exists a gap.

At last, here is my confusion: 1. Why does opencompass use so weird way to name [humaneval template](https://github.com/open-compass/opencompass/tree/main/configs/datasets/humaneval). 2. It is hard and time-consuming for users to tell which template is correct(no detail about it in default config file either). It is not friendly.

**Reply by jingmingzhuo** [reply, marked as answer]
- The prompt in humaneval_gen_4a6eef is the same one in the official repo of wizardcoder with may align with the training phase.
- The suffix represents the md5 encoding of the prompt. 
- We apologize for this confusion and will provide greater clarity.

**Reply by PJY-coder** [reply]
Thanks very much！No further question.
